### ISSUE TITLE
Dates become disabled when toggling month

### ISSUE THREAD

**Issue by nathanjhastings** [krystalcampioni/vue-hotel-datepicker#255]
### Description

Issue seems to only occur when you have a picker that is always visible, has a single day selection and starts with a default value (see code sample). Our use case is that the user may select the date, progress through the booking process then go back to change the date.

### Code sample

```vue
<HotelDatePicker 
    :alwaysVisible="true"
    :singleDaySelection="true"
    :startingDateValue="new Date('2021-06-14')"
/>
```

**Expected behavior:**

Picker defaults to startingDateValue and user is able to select a date from a different month. 

**Actual behavior:**

Picker defaults to startingDateValue but when the user toggles the month (back or forward) all other dates become disabled.

### Datepicker Version

4.0.0-beta.9

**Comment by matiasperrone** [comment]
May check the issue in beta 10?

**Comment by matiasperrone** [comment]
I will check this issue before release the next beta (14)

**Comment by nathanjhastings** [comment]
I have tested and replicated the problem on beta 15.

**Comment by matiasperrone** [comment]
Fixed in v4.0.0-RC1
